src/Sonarr.Api.V3/Qualities/QualityDefinitionResourceValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using FluentValidation;
+using NzbDrone.Core.Qualities;
+
+namespace Sonarr.Api.V3.Qualities;
+
+public class QualityDefinitionResourceValidator : AbstractValidator<QualityDefinitionResource>
+{
+    public QualityDefinitionResourceValidator(QualityDefinitionLimits limits)
+    {
+        When(c => c.MinSize is not null, () =>
+        {
+            RuleFor(c => c.MinSize)
+                .GreaterThanOrEqualTo(limits.MinLimit)
+                .WithErrorCode("GreaterThanOrEqualTo")
+                .LessThanOrEqualTo(c => c.PreferredSize ?? limits.MaxLimit)
```

Comment:
In your test, what was `MinSize` set to?

Reply:
Maybe your point here was: "I changed Preferred but it complained about Min instead". In that case, I don't think the API can really know what the user's *intent* was; especially given how the payload comes in right now. We get 3 min,preferred,max values. If I add validation to Preferred, you'll get two validation errors: One saying that Min must be lower than preferred, and one saying Preferred cannot be lower than Min (or something along those lines).

I originally had a validation rule for `PreferredSize` but this made the logic confusing *and* the test more complicated, without really adding any additional safeguards over just the min & max validation.

I agree that the verbiage is confusing; I just don't know how to make that better without overcomplicating the validation itself.

Reply:
What happened was MinSize was 20, I changed MaxSize to 5 to see how things would fail, but because the slider kept preferred -5 from max it actually set preferred to 0.

Perhaps an easier comparison would be.

```
Min >=0
Max <= 1000
Preferred >Min or 0
Preferred< Max or 1000
```

It's certainly not perfect and we don't know the intent of the user so they could submit something that gives a weird result. but at least it shouldn't try to return an error that would lead to another error.

Reply:
This use case confuses me, I'm sorry. Can you walk me through the actual steps you took? If I set Min size to 20 in the UI, and try to manually type in `5` for Max, it caps it to `21`. So I don't actually what exactly you're testing.

What exact min/preferred/max payload for that specific quality came into the API?

Reply:
I think you can do it by setting min 0, max to 5, then preferred to (if it doesn't do it automatically), then increase min and max. It's definitely a little funky doing it in the UI, but the same is true if you just hit the API directly as well.

Reply:
In my last comment I asked for the payload that was sent to the API in your test case. If you can get me that, I'll just test directly to the API instead of messing with the UI.

Reply:
I went ahead and added validation for `PreferredSize` (I originally added that to begin with but removed it thinking it was redundant). I still didn't fully understand your scenario, but I do recognize that if min & max are `null` and we have an invalid preferred size, such as `-20`, the validation wouldn't catch that without this change.